app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/EventDetailsFragment.java

```diff
@@ -178,18 +173,20 @@ private void addToTUMCalendar() {
         intent.putExtra(Const.EVENT_EDIT, false);
         intent.putExtra(Const.EVENT_TITLE, event.getTitle());
         intent.putExtra(Const.EVENT_COMMENT, event.getDescription());
-        intent.putExtra(Const.EVENT_START, DateTimeUtils.INSTANCE.getDateTimeString(event.getDate()));
-        intent.putExtra(Const.EVENT_END, DateTimeUtils.INSTANCE.getDateTimeString(
-                event.getDate().plus(eventDuration)));
+        intent.putExtra(Const.EVENT_START, DateTimeUtils.INSTANCE.getDateTimeString(event.getStart()));
+        intent.putExtra(Const.EVENT_END, DateTimeUtils.INSTANCE.getDateTimeString(event.getEnd() != null
+                ? event.getEnd()
+                : event.getStart().plus(Const.EVENT_DEFAULT_DURATION)));
         startActivity(intent);
     }
 
     private void addToExternalCalendar() {
         Intent intent = new Intent(Intent.ACTION_INSERT)
                 .setData(CalendarContract.Events.CONTENT_URI)
-                .putExtra(CalendarContract.EXTRA_EVENT_BEGIN_TIME, event.getDate().getMillis())
-                .putExtra(CalendarContract.EXTRA_EVENT_END_TIME, event.getDate().
-                        plus(eventDuration).getMillis())
+                .putExtra(CalendarContract.EXTRA_EVENT_BEGIN_TIME, event.getStart().getMillis())
+                .putExtra(CalendarContract.EXTRA_EVENT_END_TIME, event.getEnd() != null
+                        ? event.getEnd()
+                        : event.getStart().plus(Const.EVENT_DEFAULT_DURATION).getMillis())
```

Comment:
Just don't show the end date, in case it is not set.

Reply:
Well, at least for the TUM Calendar, when I look at the "initStartEndDates" method it seems like both "start" and "end" have to be set to something. For the google calendar, you could leave the extra away and it would use a default duration of one our, but I would just provide the same default duration there and keep it consistens with the tum calendar.

Reply:
I see, that makes sense. A comment for that would've been nice, but it should be alright.